woodwork/type_sys/utils.py

```diff
@@ -16,30 +14,24 @@ def col_is_datetime(col, datetime_format=None):
     if _is_spark_series(col):
         col = col.to_pandas()
 
-    if col.dtype.name.find("datetime") > -1 or (
-        len(col) and isinstance(col.head(1), datetime)
-    ):
+    if pd.api.types.is_datetime64_any_dtype(col):
         return True
 
-    # if it can be cast to numeric, it's not a datetime
+    col = col.dropna()
+    if len(col) == 0:
+        return False
+
     try:
-        pd.to_numeric(col, errors="raise")
-    except (ValueError, TypeError):
-        # finally, try to cast to datetime
-        if col.dtype.name.find("str") > -1 or col.dtype.name.find("object") > -1:
-            try:
-                pd.to_datetime(
-                    col,
-                    errors="raise",
-                    format=datetime_format,
-                    infer_datetime_format=True,
-                )
-            except Exception:
-                return False
-            else:
-                return True
+        pd.to_datetime(
```

Comment:
@jeff-hernandez Does this updated logic with the string conversion work as expected if the input series contains missing values like `None`, `np.nan` or `pd.NA`?

Would this series be recognized as a datetime series?
```
series = pd.Series(["12/01/2022", pd.NA])
```

By having the `.astype(str)` call inside the try/except block this could fail because the `pd.to_datetime` conversion fails or because the `.astype(str)` call fails.